Makefile

```diff
@@ -6,15 +6,31 @@ CROSS ?= linux/arm linux/arm64 linux/ppc64le linux/s390x \
 SUDO ?= sudo -n
 
 .PHONY: all
-all: lint test cross
+all: clean lint test test-local cross
+
+.PHONY: clean
+clean:
+	$(RM) mount/go-local.*
 
 .PHONY: test
 test: RUN_VIA_SUDO = $(shell $(SUDO) true && echo -exec \"$(SUDO)\")
-test:
+test: clean
 	for p in $(PACKAGES); do \
 		(cd $$p && go test $(RUN_VIA_SUDO) -v .); \
 	done
 
+# test the mount module against the local mountinfo source code instead of the
+# release specified in go.mod. This allows catching regressions / breaking
+# changes in mountinfo.
+.PHONY: test-local
+test-local: RUN_VIA_SUDO = $(shell $(SUDO) true && echo -exec \"$(SUDO)\")
+test-local: clean
+	echo "replace github.com/moby/sys/mountinfo => ../mountinfo" | cat mount/go.mod - > mount/go-local.mod \
+	&& cd mount \
+	&& go test -modfile=go-local.mod $(RUN_VIA_SUDO) -v . ;\
+	rm go-local.*
+	cd ../
```

Comment:
I must admit that I've been going back-and-forth what the cleanest way is to implement this;

- use `mktemp -d` to create a temporary directory for the `go-local.mod` ?
- note that bash "pseudo files" (`<(echo "replace github.com/moby/sys/mountinfo => ../mountinfo" | cat mount/go.mod -)`) don't work; `go` _REQUIRES_ a file, and it must have a `.mod` extension (possibly something we should request to be changed, so that, e.g. `STDIN` / `-` could be supported?
- combine this with the `test:` target? (less flexible, but perhaps cleaner?)
- split creation of `mount/go-local.md` to a separate target?

Input/suggestions welcome; my initial goal here is to have _some_ way to test against the local source 😅